cloudinit/sources/DataSourceOracle.py

```diff
@@ -297,32 +298,40 @@ def _add_network_config_from_opc_imds(self, set_primary: bool = False):
                 )
                 continue
             name = interfaces_by_mac[mac_address]
-            prefix = ipaddress.ip_network(
-                vnic_dict["subnetCidrBlock"]
-            ).prefixlen
+            network = ipaddress.ip_network(vnic_dict["subnetCidrBlock"])
 
             if self._network_config["version"] == 1:
-                subnet = {
-                    "type": "static",
-                    "address": f"{vnic_dict['privateIp']}/{prefix}",
+                interface_config = {
+                    "name": name,
+                    "type": "physical",
+                    "mac_address": mac_address,
+                    "mtu": MTU,
+                    "subnets": [
+                        {
+                            "address": (
+                                f"{vnic_dict['privateIp']}/{network.prefixlen}"
+                            )
+                        }
+                    ],
                 }
```

Comment:
This leads to us providing invalid network v1 on the primary nic which would have the following config:
```
subnets:
    address: <addr>/<prefix>
    type: dhcp
```
cloud-init currenty permits and silently ignores the address line on `type: dhcp` which results in no functional diff in the rendered netplan yaml, but eventually our schema validation for network config would warn/error on this.

What do you think about defining subnet in one place to avoid the supplemental mutation of interface_config down below so we have a cohesive view of primary vs not-primary device config?

```diff
diff --git a/cloudinit/sources/DataSourceOracle.py b/cloudinit/sources/DataSourceOracle.py
index ab51732f8..0b22baee2 100644
--- a/cloudinit/sources/DataSourceOracle.py
+++ b/cloudinit/sources/DataSourceOracle.py
@@ -301,23 +301,23 @@ class DataSourceOracle(sources.DataSource):
             network = ipaddress.ip_network(vnic_dict["subnetCidrBlock"])
 
             if self._network_config["version"] == 1:
+                if is_primary:
+                    # Primary interface uses DHCP as is our first entry
+                    subnet = {"type": "dhcp"}
+                else:
+                    subnet = {
+                        "type": "static",
+                        "address": (
+                            f"{vnic_dict['privateIp']}/{network.prefixlen}"
+                        ),
+                    }
                 interface_config = {
                     "name": name,
                     "type": "physical",
                     "mac_address": mac_address,
                     "mtu": MTU,
-                    "subnets": [
-                        {
-                            "address": (
-                                f"{vnic_dict['privateIp']}/{network.prefixlen}"
-                            )
-                        }
-                    ],
+                    "subnets": [subnet],
                 }
-                # Primary interface uses DHCP as is our first entry
-                interface_config["subnets"][0]["type"] = (
-                    "dhcp" if is_primary else "static"
-                )
                 self._network_config["config"].append(interface_config)
             elif self._network_config["version"] == 2:
                 # Why does this elif exist???
diff --git a/tests/unittests/sources/test_oracle.py b/tests/unittests/sources/test_oracle.py
index 274a2e282..c2be3a952 100644
--- a/tests/unittests/sources/test_oracle.py
+++ b/tests/unittests/sources/test_oracle.py
@@ -307,7 +307,6 @@ class TestNetworkConfigFromOpcImds:
             assert "02:00:17:05:d1:db" == primary_cfg["mac_address"]
             assert 9000 == primary_cfg["mtu"]
             assert 1 == len(primary_cfg["subnets"])
-            assert "10.0.0.230/24" == primary_cfg["subnets"][0]["address"]
             assert "dhcp" == primary_cfg["subnets"][0]["type"]
         secondary_cfg = nic_cfg[secondary_nic_index]
         assert "ens4" == secondary_cfg["name"]
```